hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestBlockPlacementPolicyRackFaultTolerant.java

```diff
@@ -167,6 +179,107 @@ private void doTestChooseTargetSpecialCase() throws Exception {
     }
   }
 
+  /**
+   * Verify decommission a dn which is an only node in its rack.
+   */
+  public void testPlacementWithOnlyOneNodeInRackDecommission() throws Exception {
```

Comment:
Instead of adding `testPlacementWithOnlyOneNodeInRackDecommission()` to `testChooseTarget()`, it would be better to create `testPlacementWithOnlyOneNodeInRackDecommission()` as one unit test method.
```suggestion
  @Test
  public void testPlacementWithOnlyOneNodeInRackDecommission() throws Exception {
```